securedrop/tests/conftest.py

```diff
@@ -116,44 +123,50 @@ def config(tmpdir):
 
 
 @pytest.fixture(scope='function')
-def alembic_config(config):
+def alembic_config(config: SDConfig) -> str:
     base_dir = path.join(path.dirname(__file__), '..')
     migrations_dir = path.join(base_dir, 'alembic')
     ini = configparser.ConfigParser()
     ini.read(path.join(base_dir, 'alembic.ini'))
 
     ini.set('alembic', 'script_location', path.join(migrations_dir))
-    ini.set('alembic', 'sqlalchemy.url', 'sqlite:///' + config.DATABASE_FILE)
+    ini.set('alembic', 'sqlalchemy.url', config.DATABASE_URI)
 
     alembic_path = path.join(config.SECUREDROP_DATA_ROOT, 'alembic.ini')
-    config.TESTING_ALEMBIC_PATH = alembic_path
-
     with open(alembic_path, 'w') as f:
         ini.write(f)
 
     return alembic_path
 
 
 @pytest.fixture(scope='function')
-def source_app(config):
+def source_app(config: SDConfig) -> Iterator[Flask]:
     app = create_source_app(config)
     app.config['SERVER_NAME'] = 'localhost.localdomain'
     with app.app_context():
         db.create_all()
-        yield app
+        try:
+            yield app
+        finally:
+            db.session.rollback()
+            db.drop_all()
```

Comment:
This clears the DB after each test. Otherwise, one failing test would leave data which would then cause subsequent tests to fail (for example if the tests count how many Journalists are in the DB after an action).

Reply:
I think that's a good change, though it doesn't seem necessary for the purpose of adding type annotations here.

Reply:
One my changes broke a test, but then I had a lot of other tests failing because the DB wasn't cleared between each test, so I had to add this to get to the "truly broken" test.